src/poliastro/threebody/cr3bp_char_quant.py

```diff
@@ -0,0 +1,236 @@
+"""
+@author: Dhruv Jain, Multi-Body Dynamics Research Group, Purdue University
+
+Useful ideas:
+1. To non-dimensioanlize a physical position vector: divide by l*
+2. To dimensioanlize a [nd] position vector: multiply by l*
+3. To non-dimensionalize a physical veloctiy vecotr, multiply by t*/l*
+4. To dimensionalize a [nd] veloctiy vecotr, multiply by l*/t*
+5. Position of P1 = [-mu, 0, 0] [nd] {x,y,z}
+5. Position of P2 = [1-mu, 0, 0] [nd] {x,y,z}
+"""
+from astropy import units as u
+
+import poliastro.constants.general as general
+import poliastro.constants.mean_elements as dist
+
+
+class SystemChars:
+    """Computes and stores the properties (mu, l*, t*) of a CR3BP system
+    'mu': mass ratio of P1-P2 primary bodies
+    'l*': characterisitic lenght of P1-P2 primary bodies
+    't*': characterisitic time of P1-P2 primary bodies
+    If P2 is more massive than P1 then swap the Pi, so that P1 is the more massive body
+    """
+
+    def __init__(self, p1="Earth", p2="Moon"):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        p1 : string
+           'body_i'. The deafult is Earth
+        p2 : string
+           'body_j'. The deafult is Moon
+        """
+
+        mu, lstar, tstar, p1, p2 = self.BodiesCharCompute(
+            p1.lower(), p2.lower()
+        )  # lower() to ensure lowercase characters
+        units_nd = u.def_unit(
+            "[nd]"
+        )  # Define custome units, [nd]: non-dimensional unit
+
+        self._mu = mu * units_nd
+        self._lstar = lstar
+        self._tstar = tstar
+        self._p1 = p1
+        self._p2 = p2
+
+    # All the attributes are made private to make them constant and avoid being mistakenly changed
+    @property
+    def mu(self):
+        """Mass ration of P1-P2 primary bodies in CR3BP"""
+        return self._mu
+
+    @property
+    def lstar(self):
+        """Characterisitc Length"""
+        return self._lstar
+
+    @property
+    def tstar(self):
+        """Characterisitc Time"""
+        return self._tstar
+
+    @property
+    def p1(self):
+        """Primary body P1"""
+        return self._p1
+
+    @property
+    def p2(self):
+        """Primary body P2"""
+        return self._p2
+
+    def __mu_calc(self, mu_pi):
```

Comment:
Any reason to make protected methods? Notice that private methods in Python don't really exist (`_single_underscore` methods can be called anyway) and `__double_underscore` methods exist to avoid collisions with child classes, which is something we don't expect to have here.

Reply:
The whole idea was to make the quantities constants but I think a better might be by using astropy -> Constants (). But I will appreciate if you have any other ideas on making mu, l* and t* constants.

Reply:
I'd say astropy Constants are for global constants. The Python pattern would be to make them private properties:

```python
class System:
    def __init__(self, mu, lstar, tstar):
        self._mu = mu
        self._lstar = lstar
        self._tstar = tstar

    @property
    def mu(self):
        return self._mu

    # Same for lstar, tstar

s = System(1, 1, 1)
print(s.mu)  # 1
s.mu = 1  # AttributeError: can't set attribute 'mu'
```

we don't need to go more "constant" than that I believe.